zkevm-circuits/src/evm_circuit/util/math_gadget/is_equal.rs

```diff
@@ -0,0 +1,116 @@
+use super::CachedRegion;
+use crate::evm_circuit::{util::constraint_builder::ConstraintBuilder, util::math_gadget::*};
+use eth_types::Field;
+use halo2_proofs::plonk::{Error, Expression};
+
+/// Returns `1` when `lhs == rhs`, and returns `0` otherwise.
+#[derive(Clone, Debug)]
+pub struct IsEqualGadget<F> {
+    is_zero: IsZeroGadget<F>,
+}
+
+impl<F: Field> IsEqualGadget<F> {
+    pub(crate) fn construct(
+        cb: &mut ConstraintBuilder<F>,
+        lhs: Expression<F>,
+        rhs: Expression<F>,
+    ) -> Self {
+        let is_zero = IsZeroGadget::construct(cb, lhs - rhs);
+
+        Self { is_zero }
+    }
+
+    pub(crate) fn expr(&self) -> Expression<F> {
+        self.is_zero.expr()
+    }
+
+    pub(crate) fn assign(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        lhs: F,
+        rhs: F,
+    ) -> Result<F, Error> {
+        self.is_zero.assign(region, offset, lhs - rhs)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::test_util::*;
+    use super::*;
+    use eth_types::*;
+    use gadgets::util::sum;
+    use halo2_proofs::halo2curves::bn256::Fr;
+    use halo2_proofs::plonk::Error;
+
+    #[test]
+    fn test_isequal() {
+        #[derive(Clone)]
+        /// a == b
+        struct IsEqualGadgetTestContainer<F> {
+            eq_gadget: IsEqualGadget<F>,
+            a: util::Word<F>,
+            b: util::Word<F>,
+        }
+
+        impl<F: Field> MathGadgetContainer<F> for IsEqualGadgetTestContainer<F> {
+            const NAME: &'static str = "IsEqualGadget";
+
+            fn configure_gadget_container(cb: &mut ConstraintBuilder<F>) -> Self {
+                let a = cb.query_rlc();
+                let b = cb.query_rlc();
+                let eq_gadget =
+                    IsEqualGadget::<F>::construct(cb, sum::expr(&a.cells), sum::expr(&b.cells));
```

Comment:
This looks a bit strange, the gadget works on simple field elements so wouldn't expect words to be used here.

Reply:
Changed to cell instead of RLC.